// mylib/db.js
var mongoose = require('mongoose');

exports.User = mongoose.model('User', mongoose.Schema({
    username: String,
    password: String,
    fullName: String,
    nickName: String,
    email: String,
    role: String
})
);



exports.connect = function() {
    var dbUrl;
    if (global.isProd) {
        dbUrl = process.env.OPENSHIFT_MONGODB_DB_URL + process.env.OPENSHIFT_APP_NAME;
    } else {
        dbUrl = 'mongodb://localhost/mydb';
    }
    
     mongoose.connect(dbUrl, function(err) {
            if (err) {
                console.log("[Error] MongoDB server might not be running." + err);
            }
        });
}


exports.initData = function() {
    var User = exports.User;
    
    User.findOne({username: 'admin'}, function(err, user) {
        if (err) {console.log("Error finding admin user: " + err); return;}
        if (!user) {
            var userAdmin = new User({username: 'admin', password: 'admin', fullName: 'Admin Admin', nickName: 'Admin', email: 'adminemail@email.com', role: 'ADMIN'})  
            console.log("Admin user not found, creating one");
            userAdmin.save(function(err){
            if (err) {
                console.log("Error saving admin to db: " + err);
            } else {
                console.log("Done creating admin. Username: admin , password: admin . Be sure to change this.");
            }
            });
        } else {
            //console.log("Admin user already exists.");
        }
    })
}
